frontend/packages/dev-console/src/components/pipelines/pipeline-form/MultipleResourceKeySelector.tsx

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { useField, useFormikContext, FormikValues } from 'formik';
+import cx from 'classnames';
+import * as fuzzy from 'fuzzysearch';
+import * as _ from 'lodash';
+import { RootState } from '@console/internal/redux';
+import { getActiveNamespace } from '@console/internal/reducers/ui';
+import { K8sKind, K8sResourceKind } from '@console/internal/module/k8s';
+import { ResourceDropdown, getFieldId } from '@console/shared';
+import { FormGroup } from '@patternfly/react-core';
+import {
+  useK8sWatchResource,
+  WatchK8sResource,
+} from '@console/internal/components/utils/k8s-watch-hook';
+import MultipleKeySelector from './MultipleKeySelector';
+
+interface MultipleResourceKeySelectorProps {
+  name: string;
+  label: string;
+  resourceModel: K8sKind;
+  fullWidth?: boolean;
+}
+
+interface StateProps {
+  namespace: string;
+}
+
+const MultipleResourceKeySelector: React.FC<StateProps & MultipleResourceKeySelectorProps> = ({
+  fullWidth,
+  name,
+  label,
+  namespace,
+  resourceModel,
+}) => {
+  const { setFieldValue, setFieldTouched } = useFormikContext<FormikValues>();
```

Comment:
Yes, correct. But I was thinkng of making this component independent of the form values since they can be different for different use cases.There can be some use case where the form values doesnt need to have a namespace prop so just for making this component work they will have to add a namespace prop in there form values. WDYT?

Reply:
Personally our forms and the sections within already make huge assumptions on the state of the Formik values. I'd recommend that we just go the simple route (use the form data) for now and worry about it on the flip side.

Reality of the situation is we should have FormikValues typed properly on every form and a subset available on every section. But since we don't, it makes assumptions on getting validation at a higher level and even the `name` fields lining up. So I think it's okay for now to just avoid the extra boilerplate.

With all that said, if you have done nothing, perhaps we do nothing. Let's just aim to get things merged at this point and both solutions have validity.